src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -117,9 +204,41 @@ struct _WSLA_SESSION_SETTINGS {
 ]
 interface IWSLASession : IUnknown
 {
+    // Image management.
+    HRESULT PullImage([in] LPCWSTR Image, [in, unique] struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION* RegistryInformation, [in, unique] IProgressCallback* ProgressCallback);
+    HRESULT ListImages([out, size_is(, *Count)] struct WSLA_IMAGE_INFORMATION** Images, [out] ULONG* Count);
+    HRESULT DeleteImage([in] LPCWSTR Image);
+
+    // Container management.
+    HRESULT CreateContainer([in] struct WSLA_CONTAINER_OPTIONS* Options);
+    HRESULT StartContainer([in] LPCSTR Name);
+    HRESULT StopContainer([in] LPCSTR Name);
+    HRESULT DeleteContainer([in] LPCSTR Name);
+    HRESULT GetContainerState([in] LPCSTR Name, [out] enum WSLA_CONTAINER_STATE* State);
+
+    // Anonymous host port allocation.
+    HRESULT AllocateHostPort([in] LPCSTR Name, [in] USHORT ContainerPort, [out] USHORT* AllocatedHostPort);
```

Comment:
For now the API supports both allocating ports as a container starts (via WSLA_CONTAINER_OPTIONS::Ports) and allocating ports via anonymous bind using this method.

Reply:
Hmm, what if someone leaves the WindowsPort field empty in the WSLA_PORT_MAPPING struct... does it become the anonymous bind situation then? What if we don't provide a special API to support anonymous bind, but simply have a GetHostPort() API?

Reply:
> Hmm, what if someone leaves the WindowsPort field empty in the WSLA_PORT_MAPPING struct... does it become the anonymous bind situation then?

We could potentially design something like, but in the current API state, the caller would have no way to tell which port was actually allocated (which is the reason for the dedicated method)

> simply have a GetHostPort()

That's tricky because that method needs to "reserve" the port, but if it does, then the port isn't available to WSLA anymore.

Reply:
Discussed offline: Anonymous port binding will be a P1, not needed for public preview